### PR TITLE
Add resources context manager to set CPUs/GPUs for task

### PR DESCRIPTION
**Pull request by ahuang11**
This PR allows users to [specify required resources](https://docs.ray.io/en/latest/ray-core/tasks/resources.html#), like CPUs and GPUs, for a task.

To accomplish this, we collect resources kwargs from a user through a context manager (see example below), and then pass this to `ray.remote` internally in `submit` [as suggested by Michael.](https://github.com/PrefectHQ/prefect-ray/issues/44#issuecomment-1266254662).

Closes https://github.com/PrefectHQ/prefect-ray/issues/44

### Example

Use 4 CPUs and 2 GPUs for the `process` task:
```python
from prefect import flow, task
from prefect_ray.task_runners import RayTaskRunner
from prefect_ray.context import resources
@task
def process(x):
    return x + 1
@flow(task_runner=RayTaskRunner())
def my_flow():
    # equivalent to setting @ray.remote(num_cpus=4, num_gpus=2)
    with resources(num_cpus=4, num_gpus=2):
        process.submit(42)
```

### Checklist

- [x] This pull request references any related issue by including "Closes #<ISSUE_NUMBER>"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/{{ cookiecutter.github_organization }}/{{ cookiecutter.collection_name }}/issues/new/choose) first.
- [x] This pull request includes tests or only affects documentation.
- [x] Summarized PR's changes in [CHANGELOG.md](https://github.com/{{ cookiecutter.github_organization }}/{{ cookiecutter.collection_name }}/blob/main/CHANGELOG.md)

### CODE REVIEW

File: prefect_ray/context.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        def my_flow():
+            # equivalent to setting @ray.remote(num_cpus=4, num_gpus=2)
+            with resources(num_cpus=4, num_gpus=2):
+                process.submit(42)
+        ```
+    """
+    current_resources = ResourcesContext.get().current_resources
+    new_resources.update(**current_resources)
+    with ResourcesContext(current_resources=new_resources):
+        yield
```
Comment (desertaxle) [inline prefect_ray/context.py:60]: With the current implementation, the current resources will override any new resources that are provided. I would expect the new resources to take precedence, like in the suggestion. If there's a reason for the current resources to take precedence, let me know, but as a user, I'd expect the most recent value to take precedence if I'm nesting context manager.

A test validating the chosen functionality would also be a good idea.
```suggestion
    with ResourcesContext(current_resources={**current_resources, **new_resources}):
        yield
```

**Review by desertaxle — CHANGES_REQUESTED** [review]

File: README.md [hunk 2]
```diff
@@ -193,6 +193,29 @@ pip install s3fs
...
+
+@task
+def process(x):
+    return x + 1
+
+
+@flow(task_runner=RayTaskRunner())
+def my_flow():
+    # equivalent to setting @ray.remote(num_cpus=4, num_gpus=2)
+    with resources(num_cpus=4, num_gpus=2):
```
Comment (madkinsz) [inline README.md:215]: I would name this `remote_options` instead of `resources`. It seems useful to pass other `remote` parameters.

**Review by desertaxle — APPROVED** [review]
LGTM!

File: prefect_ray/context.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        return cls.__var__.get(RemoteOptionsContext())
+
+    __var__ = ContextVar("remote_options")
+
+
+@contextmanager
+def remote_options(**new_remote_options: Dict[str, Any]) -> Dict[str, Any]:
+    """
+    Context manager to add remote_options to flow and task run calls.
+    RemoteOptions are always combined with any existing remote_options.
```
Comment (madkinsz) [inline prefect_ray/context.py:37]:
```suggestion
    Context manager to add keyword arguments to Ray `@remote` calls
    for task runs. If contexts are nested, new options are merged with options
    in the outer context. If a key is present in both, the new option will be used.
```

**Review by madkinsz — APPROVED** [review]

**Pull request merged**